### PR TITLE
CAL-15 Added OrderMgr interface for support of simple product retrieval

### PR DESCRIPTION
**Pull request by troymohl**
#### What does this PR do?
Adds the OrderMgr STANAG-4559 interface.
Added property to control whether or not to perform local or enterprise catalog searches. Default to local.
#### Who is reviewing it?
@jaymcnallie @bdeining @dcruver 
#### How should this be tested?
JUnit test cases for OrderMgr
#### What are the relevant tickets?
https://codice.atlassian.net/browse/CAL-15
#### Checklist:
- [ ] Documentation Updated
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/nsili/catalog-nsili-endpoint/src/test/java/org/codice/alliance/nsili/endpoint/TestOrderMgrImpl.java [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+        List<Result> results = new ArrayList<>(testTotalHits);
+        MetacardImpl testMetacard = new MetacardImpl();
+        testMetacard.setId(testMetacardId);
+        Result testResult = new ResultImpl(testMetacard);
+        results.add(testResult);
+        QueryResponse testResponse = new QueryResponseImpl(null, results, testTotalHits);
+        when(mockCatalogFramework.query(any(QueryRequest.class))).thenReturn(testResponse);
+    }
+
+    private void setupOrb()
```
Comment (bdeining) [inline catalog/nsili/catalog-nsili-endpoint/src/test/java/org/codice/alliance/nsili/endpoint/TestOrderMgrImpl.java]: If setupOrb is going to be used to test each interface, you may want to consider making an abstract test class so this code doesn't have to be duplicated in each test class

Comment (troymohl) [inline catalog/nsili/catalog-nsili-endpoint/src/test/java/org/codice/alliance/nsili/endpoint/TestOrderMgrImpl.java]: Good catch. I refactored that into the TestNsiliCommon that I setup for some of those test common items like that.

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/OrderRequestImpl.java [hunk 2]
```diff
@@ -13,45 +13,217 @@
...
+                                    && resourceResponse.getResource() != null) {
+                                Resource resource = resourceResponse.getResource();
+                                ResourceContainer file =
+                                        new ResourceContainer(resource.getInputStream(),
+                                                resource.getName(),
+                                                resource.getSize(),
+                                                resource.getMimeTypeValue());
+                                files.add(file);
+
+                                // Alterations aren't support, so we will only return original content
```
Comment (bdeining) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/OrderRequestImpl.java]: nit : supported

**Comment by bdeining** [comment]
:+1: after typo fix

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/OrderRequestImpl.java (hunk 2 above)
Comment (troymohl) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/OrderRequestImpl.java]: Fixed.

**Comment by dcruver** [comment]
👍 lgtm

**Pull request merged**